### DISCUSSION TITLE
Increase quota for Messages

### DISCUSSION THREAD

**Discussion by ryanbuening** [bluecaret/carettab#322] [Ideas and Feature Requests]
When you exceed 8 Kilobytes in a single item in Chrome [storage.sync](https://developer.chrome.com/docs/extensions/reference/storage/#property-sync), an error is thrown: `QUOTA_BYTES_PER_ITEM quota exceeded`.

![image](https://user-images.githubusercontent.com/812974/232130867-4a402c81-2f9e-4e4f-ad0d-f100097f7a80.png)

Nothing is shown to the user that an error has occurred, other than looking in the console. The low hanging fruit might be to at least inform the user that their Message was not saved.

A possible workaround/fix is to use the value of `QUOTA_BYTES_PER_ITEM` to split an item up into multiple items as shown in this Stack Overflow post [here](https://stackoverflow.com/questions/13373187/can-i-increase-quota-bytes-per-item-in-chrome/73558894#73558894). That solution claims to work with Manifest V3.

I understand you are working on a refactor, so maybe something to think about for that. Thanks!

**Comment by bluecaret** [comment]
That's a genius solution that could help me in a lot of other areas as well! Thank you so much for sharing!

**Comment by ryanbuening** [comment]
@bluecaret the main reason for my original post was the desire to save more messages in the Custom message > Random message feature. It looks like the title of this discussion is specific to adding an error message when exceeding the quota, which I agree would be nice improvement. 

Are you open to the idea of increasing the quota for the Random message feature? And if so, would you want a new discussion topic created for that?

Thanks.

**Reply by bluecaret** [reply]
Hi @ryanbuening Unfortunately the quota is restricted by chrome, however there are a few tricks I want to apply including what you suggested to increase the number available. There are other limits though as well, so even splitting out each message I'm still limited by the number of objects.


I might add an option similar to the notepad where you have the option of not syncing the messages with your Google account, and instead save them locally instead, this would give me a lot more space to work with, but then it wouldn't apply across computers.

I'm also looking into a potential option of saving to an external database instead which would resolve the limitations, but that has other complications.

tl;dr
I want to increase for sure, just need to find the best path 😃

**Reply by ryanbuening** [reply]
Makes sense! Thanks for the reply.

**Comment by bluecaret** [comment]
This will be mostly resolved in the upcoming v4.0.1 update. It should provide notices when running out, and I tried to improve the space used for the widget as much as possible, but is still limited by browser storage. Thanks!
